### PR TITLE
[nmos-cpp] Latest commit and upgrade boost to 1.77.0

### PR DESCRIPTION
**Pull request by pedro-alves-ferreira**
Specify library name and version:  **nmos-cpp/cci.20220113**

Include the latest commits upstream and upgrade boost to 1.77.0

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by pedro-alves-ferreira** [comment]
Waiting for #8856

File: recipes/nmos-cpp/all/conandata.yml [hunk 1]
```diff
@@ -6,3 +6,6 @@ sources:
   "cci.20211015":
     url: "https://github.com/sony/nmos-cpp/archive/d115866f8f1d2c1c55bb64d05d34aa7d61c039e3.tar.gz"
     sha256: "74e48497d6d7cc76a97b5a6331f88a58b2c96ce2f24b78176d3631051caceb6a"
+  "cci.20220113":
+    url: "https://github.com/sony/nmos-cpp/archive/fa1d946c5a52fa2f9ecf9edcd9446f72f1fdc6df.tar.gz"
+    sha256: "485caed25c6b31f666b063db4a50913a8104b081d44a29278aa98a23076b7b84"
```
Comment (garethsb) [inline recipes/nmos-cpp/all/conandata.yml:11]:
```suggestion
  "cci.20220114":
    url: "https://github.com/sony/nmos-cpp/archive/c36e1d7a7f52d205adc722d8580f5f78f9aff23d.tar.gz"
    sha256: "1ce14e9c54c4ee422169cebe27eead5de7ff7f8a2d7d95e5b75e3cf3cf5e0b60"
```

File: recipes/nmos-cpp/config.yml [hunk 2]
```diff
@@ -3,3 +3,5 @@ versions:
     folder: all
   "cci.20211015":
     folder: all
+  "cci.20220113":
```
Comment (garethsb) [inline recipes/nmos-cpp/config.yml:6]:
```suggestion
  "cci.20220114":
```

File: recipes/nmos-cpp/all/conanfile.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def config_options(self):
 
     def requirements(self):
```
Comment (garethsb) [inline recipes/nmos-cpp/all/conanfile.py:57]:
```suggestion
    def package_id(self):
        self.info.requires["boost"].minor_mode()

    def requirements(self):
```

File: recipes/nmos-cpp/all/conandata.yml (hunk 1 above)
Comment (pedro-alves-ferreira) [reply recipes/nmos-cpp/all/conandata.yml]: Thanks and apologies @garethsb!  I only saw your comments when I had already pushed similar changes.

**Comment by garethsb** [comment]
Ah, I think the failed build due to missing cpprestsdk is actually a success... now we just also need to bump cpprestsdk to Boost 1.77.0? #8856 didn't do that.

**Comment by garethsb** [comment]
@pedro-alves-ferreira I think you could now close/open this and fingers crossed 🤞 it'll build!

**Comment by garethsb** [comment]
That's obviously related to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66145, sigh. I guess my check for GCC 5 in the upstream code isn't quite what's needed...

**Comment by pedro-alves-ferreira** [comment]
@prince-chrismc @lasote @SSE4 @SpaceIm Would you be able to approve, please. Thanks!

File: recipes/nmos-cpp/all/conanfile.py [hunk 4]
```diff
@@ -73,12 +73,15 @@ def requirements(self):
...
         # nmos-cpp needs CMake 3.17 or higher but CCI doesn't allow version ranges
-        self.build_requires("cmake/3.21.2")
+        self.build_requires("cmake/3.22.0")
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, 11)
 
+    def package_id(self):
+        self.info.requires["boost"].minor_mode()
```
Comment (SpaceIm) [inline recipes/nmos-cpp/all/conanfile.py:83]: I'm wondering: do we also want this mode if nmos-cpp is static?

Comment (garethsb) [reply recipes/nmos-cpp/all/conanfile.py:83]: nmos-cpp does not support shared mode. If nmos-cpp is built against Boost 1.X, recipes using nmos-cpp and Boost must absolutely link against the same Boost 1.X, same as the case for cpprestsdk.

Comment (SpaceIm) [reply recipes/nmos-cpp/all/conanfile.py:83]: Indeed, there is no shared option.
So for my own understanding: which components of boost are responsible of this? It is still true if boost is static?

**Review by SpaceIm — APPROVED** [review]

File: recipes/nmos-cpp/all/conanfile.py (hunk 4 above)
Comment (garethsb) [reply recipes/nmos-cpp/all/conanfile.py:83]: Boost.ASIO was the culprit between Boost 1.76.0 and Boost 1.77.0 I believe.

Comment (pedro-alves-ferreira) [reply recipes/nmos-cpp/all/conanfile.py:83]: I don't know which specific components cause that problem - maybe @garethsb does - but that is irrespective of static or shared. The problem is that the ABI changed, so one module would generate e.g. a certain memory layout for a piece of data, while the other component was expecting a different layout.

**Review by uilianries — APPROVED** [review]

**Comment by pedro-alves-ferreira** [comment]
Hi @SpaceIm, I noticed you approved the PR just before the builds were completed, so could you please approve it again? Thanks!

**Comment by SpaceIm** [comment]
> so could you please approve it again?

Not required, the bot doesn't care I think. But I can approve again 😄

**Review by SpaceIm — APPROVED** [review]

**Comment by prince-chrismc** [comment]
It's the commit that counts 😉 https://github.com/conan-io/conan-center-index/blob/master/docs/review_process.md#rule-of-3-reviews

File: recipes/nmos-cpp/all/conanfile.py (hunk 4 above)
Comment (prince-chrismc) [reply recipes/nmos-cpp/all/conanfile.py:83]: That would makes sense, it was a huge update https://github.com/boostorg/asio/compare/boost-1.76.0...boost-1.77.0#diff-6e6a90601914ff7c360a3ae54062f6c22ee7ffc75c1102b8e6980a25afc904eaR10

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Is VERO switching it's implementation? 😱 Hopefully it's for another project 🤞 

(I loved the UI btw)

**Comment by garethsb** [comment]
> (I loved the UI btw)

Me too, and I've only seen the promo video 😁

**Comment by pedro-alves-ferreira** [comment]
Aha, makes sense that it's the commit. Thanks @SpaceIm and @prince-chrismc for clarifying this!

**Comment by pedro-alves-ferreira** [comment]
Nope, VERO's implementation is outstanding :wink:. This is for other projects. I'm glad you both like the UI. We're very proud of it!

**Pull request merged**